### PR TITLE
Support Az.Monitor 7.0.0 plain-string output shape

### PR DESCRIPTION
**Pull request by franklesniak**
## Description

This PR adds support for Azure PowerShell&#39;s Az.Monitor 7.0.0 output format, which changed how string-like fields (Status, Category, OperationName) are returned. Previously, these fields were PSLocalizedString objects with `.Value` and `.LocalizedValue` properties. Az.Monitor 7+ returns them as plain `[string]` values, which causes property access to fail under `Set-StrictMode -Version Latest`.

### Changes

1. **New helper function: `Resolve-LocalizableStringValue`**
   - Normalizes &#34;localizable string&#34; fields from Azure PowerShell output
   - Handles three input shapes: `$null`, plain `[string]`, and PSObject with `.Value`/`.LocalizedValue` properties
   - Returns a single `[string]` result suitable for downstream processing
   - Includes comprehensive unit tests covering all input shapes

2. **Updated `ConvertFrom-AzActivityLogRecord`**
   - Uses `Resolve-LocalizableStringValue` to normalize the `Status` field
   - Improved `OperationName` fallback logic: now resolves the shape first, then validates that it resembles an RBAC action ID (contains &#39;/&#39;) before using it as a fallback action
   - This prevents friendly display names (e.g., &#34;Delete role assignment&#34;) from polluting the clustering pipeline
   - Updated version and documentation

3. **Test coverage**
   - Added 5 new test contexts in `ConvertFrom-AzActivityLogRecord.Tests.ps1` covering:
     - Plain-string Status shape (Az.Monitor 7)
     - Status with only LocalizedValue (fallback behavior)
     - Null Status handling
     - Authorization-null + OperationName-as-display-name (record dropped)
     - Authorization-null + OperationName-as-action-id (accepted)
   - Added unit tests for `Resolve-LocalizableStringValue` (8 `It` blocks covering `$null`, plain string, empty string, legacy PSObject with both `.Value` and `.LocalizedValue`, `LocalizedValue`-only, present-but-null `.Value`, object with neither property, and value-type inputs)

4. **Module exports**
   - Added `Resolve-LocalizableStringValue` to the module&#39;s exported functions list

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

### General

- [x] My code follows the coding standards in `.github/instructions/`
- [x] I have commented my code where necessary, particularly in hard-to-understand areas
- [x] I have added or updated tests where appropriate
- [x] New and existing tests pass locally

### PowerShell-Specific

- [x] Pester tests pass locally (13 new test cases: 5 integration contexts plus 8 helper unit tests)
- [x] PowerShell formatting follows repository standards (OTBS)

## Additional Notes

The changes are backward compatible with older Az.Monitor versions (&lt;=6) while adding support for the new plain-string shape in Az.Monitor 7+. The logic gracefully handles both formats without requiring version detection.

https://claude.ai/code/session_0181QBfbUe3eCHdVv64U4fH8

### CODE REVIEW

**Comment by franklesniak** [comment]
@claude start review loop

File: src/Resolve-LocalizableStringValue.ps1 [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    # # $strResult = $null
+    #
+    # # A $null input returns $null rather than an empty string, so callers
+    # # can distinguish "absent" from "present but empty".
+    # .INPUTS
+    # None. You cannot pipe objects to this function.
+    # .OUTPUTS
+    # [string] The resolved text value. Returns $null when the input is
+    # $null or is an object that exposes neither .Value nor .LocalizedValue.
+    # .NOTES
```
[the remark this answers is not in the document]
Comment (franklesniak) [reply src/Resolve-LocalizableStringValue.ps1:52]: Valid concern — the description overstated the count.

### Options

| # | Option |
| --- | --- |
| A | Update the PR description to reflect the actual count (8 `It` blocks in the helper's test file, 5 new `It` blocks in the integration test file = 13 new cases total). |
| B | Add three more tests to the helper's test file to reach 11. |
| C | Leave the discrepancy. |

### Rubric (1-5)

| Criterion | A: fix description | B: add tests | C: no-op |
| --- | --- | --- | --- |
| PR description accuracy | 5 | 4 | 1 |
| Test quality contribution | 3 | 3 (padding to hit a number) | 3 |
| YAGNI / scope control | 5 | 2 | 5 |
| Effort | 4 | 2 | 5 |
| Future maintainability | 4 | 3 | 2 |
| **Total** | **21** | **14** | **16** |

**Selected: A.** Adding tests purely to hit the inaccurate number would be padding without coverage value. The 8 helper-level cases already span the contract (`$null`, plain string, empty string, both properties present, `LocalizedValue`-only, present-but-`$null` `.Value`, neither property, value-type input).

The PR description has been updated to state the real counts: 8 helper `It` blocks plus 5 new integration contexts (13 cases total). No code change required.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: tests/PowerShell/Resolve-LocalizableStringValue.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+        It "Returns an empty string as-is (distinguishable from null)" {
+            # Arrange, Act
+            $strResult = Resolve-LocalizableStringValue -InputObject ''
+
+            # Assert
+            $strResult | Should -Be ''
+            $strResult | Should -Not -Be $null
+        }
+    }
```
[the remark this answers is not in the document]
Comment (franklesniak) [reply tests/PowerShell/Resolve-LocalizableStringValue.Tests.ps1:39]: Valid concern — the `.OUTPUTS` block did not enumerate every case where the helper returns `$null`. The implementation also returns `$null` when the input object's `.Value` or `.LocalizedValue` property exists but its underlying value is `$null` (verified by the existing "Returns null rather than the empty string from a null cast" test).

### Options

| # | Option |
| --- | --- |
| A | Expand `.OUTPUTS` to enumerate all `$null`-return cases, and explicitly call out that an empty `[string]` input returns `''` (not `$null`). |
| B | Leave the docs as-is. |
| C | Change the implementation so a present-but-`$null` property returns `''` instead of `$null`. |

### Rubric (1-5)

| Criterion | A: update docs | B: no-op | C: change behavior |
| --- | --- | --- | --- |
| Contract accuracy | 5 | 2 | 4 |
| Backward compat | 5 | 5 | 1 |
| Implementation simplicity | 5 | 5 | 2 |
| Test alignment | 5 | 3 | 1 |
| Caller predictability | 5 | 3 | 3 |
| **Total** | **25** | **18** | **11** |

**Selected: A.** C would invert the existing tested contract (and the "distinguish absent from present-but-empty" design goal baked into the `$null`-input case). Docs-only change is the correct fix.

Applied in commit `0818ea5`. `.OUTPUTS` now documents all three `$null` paths (`$null` input, object with neither property, object whose property value is `$null`) and explicitly notes the empty-string input case.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: tests/PowerShell/Resolve-LocalizableStringValue.Tests.ps1 [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+            # Arrange
+            $objInput = [pscustomobject]@{ LocalizedValue = 'Administrative' }
+
+            # Act
+            $strResult = Resolve-LocalizableStringValue -InputObject $objInput
+
+            # Assert
+            $strResult | Should -Be 'Administrative'
+        }
+    }
```
[the remark this answers is not in the document]
Comment (franklesniak) [reply tests/PowerShell/Resolve-LocalizableStringValue.Tests.ps1:70]: Valid coverage gap — the existing test matrix exercised `Value = $null` but not `LocalizedValue = $null`, so the LocalizedValue fallback's null-handling behavior was only inferred from the `.OUTPUTS` contract.

### Options

| # | Option |
| --- | --- |
| A | Add one test for a PSObject whose `LocalizedValue` is present but `$null` (no `Value` property). |
| B | Add two tests: `LocalizedValue = $null` alone, and both properties set to `$null` together. |
| C | Leave as-is (contract is documented but not locked in). |

### Rubric (1-5)

| Criterion | A: 1 test | B: 2 tests | C: no-op |
| --- | --- | --- | --- |
| Covers reviewer's ask | 5 | 5 | 1 |
| Regression safety | 4 | 5 | 2 |
| YAGNI / scope | 5 | 3 | 5 |
| Effort | 5 | 3 | 5 |
| Docs alignment | 5 | 5 | 2 |
| **Total** | **24** | **21** | **15** |

**Selected: A.** The "both-null-together" case is largely subsumed by the existing `Value = $null` case (which has no `LocalizedValue` property at all, so the LocalizedValue branch also short-circuits). One focused test gets the coverage without padding.

Applied in commit `9701de2`. New `Context "When the input is a PSObject whose LocalizedValue is null"` exercises this path directly.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/Resolve-LocalizableStringValue.ps1 [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+            }
+
+            if ($InputObject -is [string]) {
+                return [string]$InputObject
+            }
+
+            # Guard PSObject access so non-PSObject primitive/value types
+            # fall through to the $null return rather than throwing under
+            # Set-StrictMode -Version Latest.
+            if ($null -ne $InputObject.PSObject -and $null -ne $InputObject.PSObject.Properties) {
```
[the remark this answers is not in the document]
Comment (franklesniak) [reply src/Resolve-LocalizableStringValue.ps1]: Valid — the "non-PSObject primitive/value types" framing was inaccurate. In PowerShell every non-`$null` value exposes a `PSObject` wrapper, so the `$null -ne $InputObject.PSObject` clause was always true once the `$null` and `[string]` fast paths above had run. The real intent was to probe for `Value`/`LocalizedValue` presence under StrictMode without throwing.

### Options

| # | Option |
| --- | --- |
| A | Accept the suggestion verbatim — simplify the guard and rewrite the comment. |
| B | Keep the guard structure, only fix the comment. |
| C | Leave both as-is. |

### Rubric (1-5)

| Criterion | A: accept | B: comment-only | C: no-op |
| --- | --- | --- | --- |
| Comment accuracy | 5 | 5 | 2 |
| Code simplicity | 5 | 3 | 3 |
| Defensive robustness | 4 | 5 | 5 |
| Maintainer clarity | 5 | 4 | 2 |
| Backward compat | 5 | 5 | 5 |
| **Total** | **24** | **22** | **17** |

**Selected: A.** The dropped `PSObject -ne $null` clause was dead code given the `[string]` early return; keeping it only added noise.

Applied in commit `9701de2`. Guard is now `if ($null -ne $InputObject.PSObject.Properties)` and the comment describes the StrictMode-probe intent directly.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/GloryRole.psd1 [hunk 5]
```diff
@@ -37,6 +37,7 @@
         'New-FeatureIndex'
         'Remove-DuplicateCanonicalEvent'
         'Remove-RareAction'
+        'Resolve-LocalizableStringValue'
         'Resolve-PrincipalKey'
```
[the remark this answers is not in the document]
Comment (franklesniak) [reply src/GloryRole.psd1:41]: Valid — the inline count went stale the moment `Resolve-LocalizableStringValue` was added. Confirmed by counting `src/*.ps1`: 29 files total, 28 function files once `Invoke-RoleMiningPipeline.ps1` (the script entry point) is excluded.

### Options

| # | Option |
| --- | --- |
| A | Update the count to 28 to match current reality. |
| B | Drop the count entirely (less maintenance burden, but deviates from the existing convention). |
| C | Leave it at 27. |

### Rubric (1-5)

| Criterion | A: update to 28 | B: drop count | C: no-op |
| --- | --- | --- | --- |
| Accuracy | 5 | 5 | 1 |
| Maintenance burden | 2 | 5 | 5 |
| Reader info value | 3 | 2 | 2 |
| Respects existing convention | 5 | 3 | 5 |
| Matches reviewer's literal ask | 5 | 3 | 1 |
| **Total** | **20** | **18** | **14** |

**Selected: A.** B is tempting on maintenance-burden grounds, but deviation from both the existing convention and the reviewer's literal ask needs stronger justification than "one fewer line to touch on the next add."

Applied in commit `9701de2`.

---
_Generated by [Claude Code](https://claude.ai/code)_
